Source/Assets/Scripts/Player/PlayerBonuses.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerBonuses : MonoBehaviour
+{
+    public Action<Bonuses, bool> OnBonusChange;
+
+    [SerializeField] private Room[] rooms = new Room[0];
+    [SerializeField] private bool verboseLogging = false;
+
+    private Dictionary<Bonuses, bool> bonuses = new Dictionary<Bonuses, bool>();
+
+    public bool this[Bonuses key] => bonuses[key];
+
+    private void Awake()
+    {
+        if (verboseLogging)
+        {
+            Debug.Log(nameof(Awake), this);
+        }
+
+        Array bonusValues = Enum.GetValues(typeof(Bonuses));
+        foreach (Bonuses bonusValue in bonusValues)
+        {
+            bonuses.Add(bonusValue, true);
+        }
+
+        foreach (Room room in rooms)
```

Comment:
You can simplify this by using static Actions. That way there is no need to store Room instances, you can just pass in the room instance itself, or ID or Bonus that it represents.